server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -170,7 +170,26 @@ public void onFailure(final String source, final Exception e) {
                                         @Override
                                         public void clusterStateProcessed(final String source,
                                                                           final ClusterState oldState, final ClusterState newState) {
-                                            listener.onResponse(new AcknowledgedResponse(acknowledged));
+
+                                            final AcknowledgedResponse response = new AcknowledgedResponse(acknowledged);
+                                            final String[] indices = results.entrySet().stream()
+                                                .filter(result -> result.getValue().isAcknowledged())
+                                                .map(result -> result.getKey().getName())
+                                                .filter(index -> newState.routingTable().hasIndex(index))
+                                                .toArray(String[]::new);
+
+                                            if (indices.length > 0) {
+                                                activeShardsObserver.waitForActiveShards(indices, request.waitForActiveShards(),
+                                                    request.ackTimeout(), shardsAcknowledged -> {
+                                                        if (shardsAcknowledged == false) {
+                                                            logger.debug("[{}] indices closed, but the operation timed out while waiting " +
+                                                                "for enough shards to be started.", Arrays.toString(indices));
+                                                        }
+                                                        listener.onResponse(response);
```

Comment:
are we ignoring the value of `shardsAcknowledged` here? I think we need to add a `CloseIndexResponse` that extends `ShardsAcknowledgedResponse` if we want to do this PR here.

Reply:
Yes, this is part of another PR I wanted to open after this one, but it makes more sense to do it in this PR and then improve the `CloseIndexResponse` in another PR.